src/app/obfboard.ts

```diff
@@ -130,16 +203,31 @@ export class Image {
 }
 
 export class Sound {
+
+  @IsNotEmpty({ message: 'Sound id must be specified'})
+  @IsString({ message: 'Sound id must be a string'})
+  @OneOf(['url', 'data', 'path'], { message: 'Sound with id "$value" must specifiy data, a url or a path' })
   id: string;
+
+  @IsOptional()
   data: string;
+
+  @IsOptional()
+  @IsUrl()
   url: string;
+
+  @IsOptional()
   path: string;
+
   contentType: string;
+
   duration: number;
+
+  @IsDefined()
   parent: OBFBoard;
 
   deserialize(input: any, parent: OBFBoard): Sound {
-    this.id = String(input.id);
+    this.id = input.id && String(input.id);
```

Comment:
```suggestion
     this.id = (input.id || input.id === 0) && String(input.id);
```
Because 0 won't cast otherwise